library/src/main/java/com/kizitonwose/calendarview/model/MonthConfig.kt

```diff
@@ -106,11 +106,7 @@ internal data class MonthConfig(
 
             // Regroup data into 7 days.
             val allDaysGroup = mutableListOf<List<CalendarDay>>()
-            while (allDays.isNotEmpty()) {
-                val sevenDays = allDays.take(7)
-                allDaysGroup.add(sevenDays)
-                allDays.removeAll(sevenDays)
-            }
+            allDaysGroup.addAll(allDays.chunked(7))
```

Comment:
This can be further simplified as:

```kotlin
// Regroup data into 7 days. Use toList() to create a copy of the ephemeral list.
val allDaysGroup = allDays.chunked(7).toMutableList()
```